envoy/ssl/context.h

```diff
@@ -20,6 +20,12 @@ class Context {
 public:
   virtual ~Context() = default;
 
+  /**
+   * @return a vector of pairs of cert name and seconds until expiration for all loaded certs.
+   */
+  virtual std::vector<std::pair<std::string, uint64_t>>
```

Comment:
Replace std::pair usages with struct for clearness to avoid reading code with `.first`, `.second`